portfolio/src/main/java/com/google/sps/servlets/ScheduleServlet.java

```diff
@@ -40,27 +43,50 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     JSONArray eventsArray = jsonFromRequest.getJSONArray("events");
     JSONArray tasksArray = jsonFromRequest.getJSONArray("tasks");
-    String workHoursStartTimeString = (String) jsonFromRequest.get("startTime");
-    String workHoursEndTimeString = (String) jsonFromRequest.get("endTime");
+    String workHoursStartTimeString = jsonFromRequest.getString("startTime");
+    String workHoursEndTimeString = jsonFromRequest.getString("endTime");
     Instant workHoursStartTime = Instant.parse(workHoursStartTimeString);
     Instant workHoursEndTime = Instant.parse(workHoursEndTimeString);
-
+    String algorithmTypeString = jsonFromRequest.getString("algorithmType");
     Collection<CalendarEvent> events = collectEventsFromJsonArray(eventsArray);
     Collection<Task> tasks = collectTasksFromJsonArray(tasksArray);
+
+    Optional<SchedulingAlgorithmType> schedulingAlgorithmTypeOptional =
+        getSchedulingAlgorithmTypeOptional(algorithmTypeString);
+    if (!schedulingAlgorithmTypeOptional.isPresent()) {
+      response.sendError(
+          HttpServletResponse.SC_BAD_REQUEST,
+          "The request by the client was syntactically incorrect.");
+      // Here I am returning the empty schedule instead of null to not mess up
+      // any front end code expecting some array.
+      returnEmptyArrayResponse(response);
+      return;
+    }
+
+    Optional<BaseTaskScheduler> algorithmOptional =
```

Comment:
Consider renaming to taskSchedulerOptional. And as I suggested above, this variable can be of type Optional<TaskScheduler> where TaskScheduler is a new interface.

Reply:
Done.